server/plugin/virtual_agent_test.go

```diff
@@ -638,6 +639,8 @@ func Test_HandlePreviousCarouselPosts(t *testing.T) {
 			p.store = mockedStore
 
 			p.handlePreviousCarouselPosts(testutils.GetID())
+			// Adding a timer to wait for all the goroutines to finish
+			time.Sleep(500 * time.Millisecond)
```

Comment:
is this feasible? I think it can do the work but worried if this should be done.